protos/table.proto

```diff
@@ -58,7 +58,33 @@ message Manifest {
     // that the library is semantically versioned, this is a string. However, if it
     // is semantically versioned, it should be a valid semver string without any 'v'
     // prefix. For example: `2.0.0`, `2.0.0-rc.1`.
+    //
+    // For forward compatibility with older readers, when writing new manifests this
+    // field should contain only the core version (major.minor.patch) without any
+    // prerelease or build metadata. The prerelease/build info should be stored in
+    // the separate prerelease and build_metadata fields instead.
```

Comment:
Let's make it clear this is a compatibility issue specific to Lance Rust library and only for certain old versions. That makes it clearer to new libraries what the impact would be of deviating from this.

```suggestion
    // Compatibility note: The Lance Rust library in versions <=0.38.3 would
    // panic if it encountered a non-semver version string here or a version string
    // with anything more than major, minor, patch version.
    // For forward compatibility with older readers, when writing new manifests this
    // field should contain only the core version (major.minor.patch) without any
    // prerelease or build metadata. The prerelease/build info should be stored in
    // the separate prerelease and build_metadata fields instead.
```